core/src/main/java/hudson/Util.java

```diff
@@ -191,16 +191,28 @@ public static String loadFile(@Nonnull File logfile, @Nonnull Charset charset) t
         if(!logfile.exists())
             return "";
 
-        StringBuilder str = new StringBuilder((int)logfile.length());
-
-        try (BufferedReader r = Files.newBufferedReader(fileToPath(logfile), charset)) {
```

Comment:
What's the motivation to change more than just this line?

Reply:
`Files.newInputStream(fileToPath(logfile))` should be its own resource to prevent the leak of an OS file handle if construction of the `InputStreamReader` or `BufferedReader` were to fail. Beyond that, I was just trying to optimize the method as much as I could given that Files.newBufferedReader() couldn't be used here. However, this morning I am rethinking this decision to change more; it's probably safer to revert only that which is necessary at this point. I have pushed another commit to restore the use of `StringBuilder` and reading into `buf`.